ARCs/arc-0003.md

```diff
@@ -92,13 +96,34 @@ The JSON Metadata schema is as follows:
 
 The property `decimals` is **OPTIONAL**. If provided, it **MUST** match the ASA parameter `dt`.
 
+#### Integrity Properties
+
+Compared to ERC-1155, the metadata JSON schema allows to indicate digests of the files pointed by any URI field.
+This is to ensure the integrity of all the files referenced by the ASA.
+Concretly, every URI property `xxx` is allowed to have an optional associated property  `xxx_integrity` that specifies the digest of the file pointed by the URI.
+
+The digests are represented as a single SHA-256 integrity metadata as defined in the [W3C subresource integrity specification](https://w3c.github.io/webappsec-subresource-integrity).
+Details on how to generate those digests can be found on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) (where `sha384` or `384` are to be replaced by `sha256` and `256` respectively as only SHA-256 is supported by this ARC).
+
+It is **RECOMMENDED** to specify all the `xxx_integrity` properties of all the `xxx` URI properties.
+
+> Note that the only URI property specified by this schema is `image`. 
+> However, other properties may be used by specific ASAs. 
+> For example, for commonly used properties `external_url`, `animation_url`, or `youtube_url`, digests of the corresponding files may be specified in properties `external_url_integrity`, `animation_url_integrity`, `youtube_url_integrity`.
+
+Any property with a name ending with `_integrity` **MUST** match a corresponding property containing a URI to a file with a matching digest.
+For example, if the property `hello_integrity` is specified, the property `hello` **MUST** exist and **MUST** be a URI pointing to a file with a digest equal to the digest specified by `hello_integrity`.
+
+#### Sample
+
 An example of an ARC-3 Metadata JSON file follows. The properties array proposes some SUGGESTED formatting for token-specific display properties and metadata.
 
 ```json
 {
 	"name": "My Picture",
 	"description": "Lorem ipsum...",
 	"image": "https:\/\/s3.amazonaws.com\/your-bucket\/images\/{id}.png",
```

Comment:
Why are the slashes escaped with backslashes?

Reply:
Apparently in JSON you can espace slashes by backslashes and you can also not do it.
The reason seems to be that when you embed JSON in HTML, you should never have `</` anywhere.
So you need to espace `/` by `\/` in that case.

The reason why this one has escapes is just copy-pasting from ERC-1155...